diff --git a/lib/environment.sh b/lib/environment.sh
index 4c31767..8dd36a3 100644
--- a/lib/environment.sh
+++ b/lib/environment.sh
@@ -77,8 +77,8 @@ debci_salsa_client_secret="${debci_salsa_client_secret:-}"
 
 debci_secrets_dir=${debci_secrets_dir:-$(readlink -f "${debci_base_dir}/secrets")}
 
-shared_short_options='c:s:a:b:d:m:hq'
-shared_long_options='config:,suite:,arch:,backend:,data-dir:,amqp:,mirror:,keyring:,help,quiet'
+shared_short_options='c:s:a:b:d:l:m:hq'
+shared_long_options='config:,suite:,arch:,backend:,data-dir:,logs-dir:,amqp:,mirror:,keyring:,help,quiet'
 
 usage_shared_options="Common options:
 
@@ -92,6 +92,7 @@ usage_shared_options="Common options:
                             (default: unstable)
   -d DIR, --data-dir DIR    the directory in which debci will store its data,
                             and where it will read from
+  -l DIR, --logs-dir DIR    the directory to store autopkgtest logs
   --amqp amqp://[user:password@]hostname[:port]
                             AMQP server to connect to (default: ${debci_amqp_server})
   -m URL, --mirror URL      selects which mirror to use for APT-related actions,
@@ -134,6 +135,9 @@ for arg in "$@"; do
       -d|--data-dir)
         var=debci_data_basedir
         ;;
+      -l|--logs-dir)
+        var=debci_logs_dir
+        ;;
       --amqp)
         var=debci_amqp_server
         ;;
@@ -222,6 +226,9 @@ debci_session_secret="${debci_session_secret:-}"
 # page to display when authentication fails
 debci_auth_fail_page="${debci_auth_fail_page:-}"
 
+# parameter for autopkgtest --logs-dir
+debci_logs_dir="${debci_logs_dir:-}"
+
 # calculate distro based on debci_suite
 __script="/usr/share/debootstrap/scripts/${debci_suite:?}"
 if [ -r "$__script" ]; then
